stl/src/print.cpp

```diff
@@ -0,0 +1,152 @@
+// Copyright (c) Microsoft Corporation.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+// print.cpp -- C++23 <print> implementation
+
+#include <__msvc_print.hpp>
+#include <cstdio>
+#include <expected>
+#include <internal_shared.h>

```

Comment:
I'm assuming that you are referring to `<internal_shared.h>`.

When you wrote that comment, I had implemented the functions defined in `print.cpp` using `std::basic_string` and `std::basic_string_view`. I initially tried implementing them using `__crt_unique_heap_ptr<wchar_t>`, much like how heap allocations are done within `filesystem.cpp`. This class template is defined inline within `<internal_shared.h>`, so I must have left that in there after experimenting.

As you suggested in a previous code review, however, we don't need to make a separate satellite DLL for `std::print()`, so this is no longer the case. Now that `print.cpp` is directly embedded within the `msvcprt[d].lib` static import libraries, however, we don't want `std::basic_string` or `std::basic_string_view` to be included, since that would result in significant binary bloat. As a result, we now *are* making use of `__crt_unique_heap_ptr<wchar_T>`.

So, in short, `<internal_shared.h>` didn't need to be included when you wrote this code review, but it does now. That's why I'm marking this as resolved without performing changes.